framework/include/cppmicroservices/BundleContext.h

```diff
@@ -668,11 +826,11 @@ class US_Framework_EXPORT BundleContext
    * @see RemoveServiceListener()
    */
   template<class R>
-  void AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
-                          const std::string& filter = std::string())
+  ListenerToken AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
```

Comment:
IMHO, having only one way to add listeners is preferable. I didn't come across any other (more or less well-known) library which would expose a second API function for convenience member function registration. `std::bind` is a common and accepted technique from what I have seen.